cedar-drt/fuzz/src/schemas.rs

```diff
@@ -0,0 +1,95 @@
+use cedar_policy_validator::{ActionType, ApplySpec, NamespaceDefinition, SchemaFragment};
+
+/// Check if two schema fragments are equivalent
+pub fn semantic_equality_check(lhs: SchemaFragment, rhs: SchemaFragment) -> Result<(), String> {
+    if lhs.0.len() == rhs.0.len() {
+        lhs.0
+            .into_iter()
+            .map(|(name, lhs_namespace)| {
+                let rhs_namespace = rhs
+                    .0
+                    .get(&name)
+                    .ok_or_else(|| format!("`{name}` does not exist in RHS schema"))?;
+                semantic_namespace_equality(lhs_namespace, rhs_namespace.clone())
+            })
+            .fold(Ok(()), result_combiner)
+    } else {
+        Err("schema differ in number of namespaces".to_string())
+    }
+}
+
+fn semantic_namespace_equality(
+    lhs: NamespaceDefinition,
+    rhs: NamespaceDefinition,
+) -> Result<(), String> {
+    if lhs.common_types != rhs.common_types {
+        Err("Common types differ".to_string())
+    } else if lhs.entity_types != rhs.entity_types {
+        Err("Entity types differ".to_string())
+    } else if lhs.actions.len() != rhs.actions.len() {
+        Err("Different number of actions".to_string())
+    } else {
+        lhs.actions
+            .into_iter()
+            .map(|(name, lhs_action)| {
+                let rhs_action = rhs
+                    .actions
+                    .get(&name)
+                    .ok_or_else(|| format!("Action `{name}` not present on rhs"))?;
+                semantic_action_type_equality(name.as_ref(), lhs_action, rhs_action.clone())
+            })
+            .fold(Ok(()), result_combiner)
```

Comment:
If I read `result_combiner` correctly:
```suggestion
            .fold(Ok(()), Result::and)
```